### PR TITLE
Update autotools use

### PR DESCRIPTION
**Pull request by rrthomas**
This gets rid of autogen.sh: autoreconf is used instead

### CODE REVIEW

**Review by hfiguiere — CHANGES_REQUESTED** [review]
Thanks for catching these documentation issues. 

But `autogen.sh` must stay. It is expected by things like OStree, etc. It is also the reason why TravisCI failed (but that could be fixed)

**Comment by rrthomas** [comment]
No problem, it can be reduced to one line, to call `autoreconf`. I shall update the PR.

**Comment by hfiguiere** [comment]
it is expected thar `autogen.sh` runs configure unless `NOCONFIGURE=1` env is set. Therefor there shouldn't be any need to change the `travis.yml` file. Also I think you should squash a few of these commits together.

**Comment by rrthomas** [comment]
I noticed that it was trying to run `make` without first running `./configure`. But I didn't set `NOCONFIGURE`.

Anyway, I will get it working first, then I will squash commits, then I will try again changing the script back to what it was originally. Thanks!

**Comment by TingPing** [comment]
A reasonable autogen.sh is something like:

```sh
#!/bin/sh
# Run this to generate all the initial makefiles, etc.

srcdir=`dirname $0`
test -z "$srcdir" && srcdir=.

(test -f $srcdir/configure.ac) || {
	echo -n "**Error**: Directory "\`$srcdir\'" does not look like the top-level directory"
	exit 1
}

aclocal --install -I m4 || exit 1
autoreconf --install -Wno-portability || exit 1

if [ "$NOCONFIGURE" = "" ]; then
        $srcdir/configure "$@" || exit 1

        if [ "$1" = "--help" ]; then exit 0 else
                echo "Now type \`make\' to compile" || exit 1
        fi
else
        echo "Skipping configure process."
fi

set +x
```

Might also need `libtoolize --quiet --copy || exit 1` for this.

**Comment by rrthomas** [comment]
What's wrong with just running `autoreconf -i`?

**Comment by rrthomas** [comment]
As far as I can see, it is up to `autogen.sh` to provide the `NOCONFIGURE` logic. The old script did not; it had a `--no-configure` command-line flag instead.

Hence, I've used @TingPing's suggestion.

**Comment by rrthomas** [comment]
All looks good now, let me know if there's anything else.

By the way, I'll need to update the libtool versioning scheme in my `enchant-2-0` branch, because the current scheme doesn't work when the major number is incremented and the minor reset to 0.

**Comment by hfiguiere** [comment]
It was an old autogen.sh that had not been updated recently. @TingPing suggestion is the right one

Part of the rationale is that we can run `autogen.sh` from a `builddir` that is not `srcdir`. This is also expected.

File: Makefile.am [hunk 1]
```diff
@@ -1,10 +1,11 @@
+ACLOCAL_AMFLAGS = -I m4
```
Comment (hfiguiere) [inline Makefile.am:1]: I think that what you are looking for is `ac-helpers`. Not sure how up to date it is, and I think it is fine to rename it to `m4`.

File: Makefile.am [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
+
 SUBDIRS=src tests doc data
 
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = enchant.pc
 
 EXTRA_DIST =		\
-	autogen.sh	\
```
Comment (hfiguiere) [inline Makefile.am:7]: You need to keep this. autogen.sh must be in the dist.

File: configure.ac [hunk 3]
```diff
@@ -1,44 +1,45 @@
-AC_INIT(src/enchant.h)
+AC_INIT([enchant],[1.6.1])
+AC_PACKAGE_VERSION
+AC_CONFIG_SRCDIR(src/enchant.h)
+AM_INIT_AUTOMAKE([no-define])
 AC_CANONICAL_HOST
+AM_MAINTAINER_MODE
+AC_CONFIG_MACRO_DIR([m4])
```
Comment (hfiguiere) [inline configure.ac:7]: see my comment above about how there is `ac-helpers`

**Review by hfiguiere — CHANGES_REQUESTED** [review]

**Comment by rrthomas** [comment]
Good catch about `EXTRA_DIST`! Thanks for the note about `ac-helpers`.

**Review by hfiguiere — APPROVED** [review]

**Pull request merged**